README.md

```diff
@@ -64,14 +74,173 @@ If you have a suggestion for a feature, please feel free to file it under "issue
+
+One of the powerful things about Script Lab is the ability to incorporate other libraries with your snippets. This creates a bunch of fun and compelling scenarios (for example, see the color-picker example, which you can import using the GIST URL <https://gist.github.com/Zlatkovsky/c7ba602e8a107cfe5dd6c42ca41deac1>)
+
+![Import](.github/images/color-picker.jpg)
+
+The ease of importing varies depending on how the external library was created. If you are not sure, start with the first option, and progress to the next one if it doesn't work for you
+
+
+#### Easiest: library is hosted on a regular CDN, exposes a global variable, and/or is a UMD module
+
+This is the "happy path".  Simply reference the library (both the JS and the CSS, if relevant) in the "Libraries" tab.
+
+If the library is hosted on a CDN (much as `Office.js` is), you could reference it as a URL, as follows:
+
+    https://appsforoffice.microsoft.com/lib/1/hosted/office.js
+
+If the library is available as an NPM package, you can just reference it by name:
+
+    spectrum-colorpicker/spectrum.js
+    spectrum-colorpicker/spectrum.css
+
+This will use <https://unpkg.com/>, a global CDN for NPM libraries, to substitute in the appropriate files. Thus, the above is equivalent to:
+
+    https://unpkg.com/spectrum-colorpicker@1.8.0/spectrum.js
+    https://unpkg.com/spectrum-colorpicker@1.8.0/spectrum.css
+
+> Tip: To see the available files in a package, simply include a `/` after the package name, and you'll get a directory listing.  E.g., <https://unpkg.com/spectrum-colorpicker/>. It will also conveniently tell you the available version numbers, via a dropdown at the top right of the site:
+>
+> ![Import](.github/images/unpkg.jpg)
+
+
+#### Moderate: NPM package is an AMD module, but the library is also available via a "dist" build
+
+Suppose you add a reference to the library (say, the [javascript-qrcode](https://www.npmjs.com/package/javascript-qrcode) NPM package), but all you get is an error in the console:
+
+    ["Script error",
+     "https://unpkg.com/javascript-qrcode",
+     0, 0, undefined]
+
+In Internet Explorer (and hence on Add-ins on a PC version of Office), you don't get much more information than this -- but on a different platform (or Office Online on a different browser), you will see:
+
+    ["Uncaught ReferenceError: module is not defined",
+     "https://unpkg.com/javascript-qrcode@1.0.7",
+     3302, 1, ReferenceError: module is not defined {}]
+
+If you are lucky, even though the *NPM* package doesn't contain a web-ready version of the package, the web-compatible version may be available elsewhere.  In the case of this `javascript-qrcode` package, the [NPM entry](https://www.npmjs.com/package/javascript-qrcode) points to a GitHub repo, <https://github.com/siciarek/javascript-qrcode>. And the repo, in turn, has a "dist" folder (convention), along with exampe usage that also shows `<script src="dist/qrcode.min.js"></script>`.
```

Comment:
Instead taking a fail first approach would help, as the developer to verify that the library is web compatible. Then try and reference via unpkg. If it still doesnt work, examine the contents of the package, try rawgit. else post an issue on the developers repository to expose an npm package.